modules/firewall-rules/variables.tf

```diff
@@ -25,28 +25,37 @@ variable "network_name" {
 }
 
 variable "rules" {
-  description = "List of custom rule definitions (refer to variables file for syntax)."
+  description = "List of custom rule definitions"
   default     = []
   type = list(object({
     name                    = string
-    description             = optional(string)
-    direction               = optional(string)
-    priority                = optional(number)
-    ranges                  = optional(list(string))
+    description             = optional(string, null)
+    direction               = optional(string, "INGRESS")
+    priority                = optional(number, null)
+    ranges                  = optional(list(string), [])
+    destination_ranges      = optional(list(string), [])
+    source_ranges           = optional(list(string), [])
```

Comment:
It's confusing to see 3 range attributes although I understand that we're trying to not create a breaking change.

What do you think about introducing two new variables i.e. `ingress_rules` & `egress_rules` instead? By default we can redirect `rules` to the appropriate "rules" var i.e. `ingress_rules` or `egress_rules` based on the direction and use the other one for opposite direction. And add to `rules` description that it is deprecated.

This way, customers using `rules` will be fine for now while new customers could specifically use `ingress_rules` and/or `egress_rules` as needed. This will eliminate the need for custom validation for `rules` as well.